File: jour29.c
#include <stdio.h>
#include <string.h>

// 1. Définition des structures
typedef struct {
    char matiere[20];
    float note;
} Matiere;

typedef struct {
    char nom[50];
    char filiere[30];
    Matiere notes[3]; // Tableau de 3 structures "Matiere"
    float moyenne;
    char statut[10];
} Etudiant;

int main() {
    Etudiant classe[100]; 
    int nbEtudiants = 0;
    int choix;

    do {
        printf("\n--- MENU GESTION C ---");
        printf("\n1. Ajouter etudiant\n2. Afficher liste\n3. Quitter\nChoix: ");
        scanf("%d", &choix);

        if (choix == 1) {
            printf("Nom: "); scanf("%s", classe[nbEtudiants].nom);
            printf("Filiere: "); scanf("%s", classe[nbEtudiants].filiere);

            float somme = 0;
            char* nomsMatieres[] = {"Maths", "Algo", "Anglais"};

            for(int i = 0; i < 3; i++) {
                strcpy(classe[nbEtudiants].notes[i].matiere, nomsMatieres[i]);
                printf("Note en %s: ", nomsMatieres[i]);
                scanf("%f", &classe[nbEtudiants].notes[i].note);
                somme += classe[nbEtudiants].notes[i].note;
            }

            classe[nbEtudiants].moyenne = somme / 3;
            strcpy(classe[nbEtudiants].statut, (classe[nbEtudiants].moyenne >= 10) ? "Admis" : "Ajourne");
            
            nbEtudiants++;
            printf("Etudiant ajoute !\n");
        } 
        else if (choix == 2) {
            printf("\n%-15s | %-15s | %-10s | %s\n", "NOM", "FILIERE", "MOYENNE", "STATUT");
            for(int i = 0; i < nbEtudiants; i++) {
                printf("%-15s | %-15s | %-10.2f | %s\n", 
                    classe[i].nom, classe[i].filiere, classe[i].moyenne, classe[i].statut);
            }
        }
    } while (choix != 3);

    return 0;
}

File: jour29.py
#fo'ction pour le calcul de la moyenne
def calculer_moyenne(liste_notes):
    if not liste_notes:
        return 0
    return sum(liste_notes) / len(liste_notes)
#fonction piir verifier admission 
def verifier_admission(moyenne):
    return "Admis" if moyenne >= 10 else "Ajourné"
#fonction pro'ciplaed 
def main():
    classe = []
    
    while True:
        print("\n--- MENU DE GESTION SCOLAIRE ---")
        print("1. Ajouter un étudiant")
        print("2. Afficher la liste des étudiants")
        print("3. Quitter")
        
        choix = input("Votre choix : ")

        if choix == "1":
            # SAISIE DES DONNeES 
            nom = input("Nom de l'étudiant : ")
            filiere = input("Filière : ")
            
            notes = []
            matieres = ["Maths", "Algo", "Anglais"]
            
            print(f"Saisie des notes pour {nom} :")
            for mat in matieres:
                note = float(input(f"  Note en {mat} : "))
                notes.append(note)
            moy = calculer_moyenne(notes)
            statut = verifier_admission(moy)
            
            #  STOCKAGE 
            etudiant = {
                "nom": nom,
                "filiere": filiere,
                "notes": notes,
                "moyenne": moy,
                "statut": statut
            }
            classe.append(etudiant)
            print(f"\n✅ Étudiant {nom} ajouté avec succès !")

        elif choix == "2":
            # AFFICHAGE DES RÉSULTATS 
            if not classe:
                print("\nLa liste est vide.")
            else:
                print("\n" + "="*60)
                print(f"{'NOM':<15} | {'FILIÈRE':<15} | {'MOYENNE':<10} | {'RÉSULTAT'}")
                print("-" * 60)
                for etu in classe:
                    print(f"{etu['nom']:<15} | {etu['filiere']:<15} | {etu['moyenne']:<10.2f} | {etu['statut']}")
                print("="*60)

        elif choix == "3":
            print("Fin du programme.")
            break
        else:
            print("Choix invalide, réessayez.")


if __name__ == "__main__":
    main()
